File: 0110-balanced-binary-tree/0110-balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int height(TreeNode* root){
        if(!root)return 0;
        int left = 1 + height(root->left);
        int right = 1 + height(root->right);
         return max(left,right);
    }
    int solve(TreeNode* root){
        if(!root)return 0;
        int left =  solve(root->left);
        if(left == -1)return -1;
        int right = solve(root->right);
        if(right==-1)return -1;
        return (abs(left-right) <= 1) ? max(left,right) + 1 : -1;
    }
    bool isBalanced(TreeNode* root) {
        if(!root)return true;
        return solve(root) != -1;
    }
};

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int solve(TreeNode root){
        if(root == null)return 0;
        int left = solve(root.left);
        int right = solve(root.right);
        if(Math.abs(left - right) > 1) return -1;
        if(left == -1 || right == -1) return -1;
        return 1 + Math.max(left, right);
    }
    public boolean isBalanced(TreeNode root) {
        if(root == null)return true;
        int ans = solve(root);
        return ans != -1;
    }
}